wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController+NavigationBar.swift

```diff
@@ -111,25 +111,45 @@ extension ConversationViewController {
     }
 
     var backButton: UIBarButtonItem {
-        let hasUnreadInOtherConversations = self.conversation.hasUnreadMessagesInOtherConversations
-        let arrowIcon: StyleKitIcon = view.isRightToLeft
-        ? (hasUnreadInOtherConversations ? .forwardArrowWithDot : .forwardArrow)
-        : (hasUnreadInOtherConversations ? .backArrowWithDot : .backArrow)
 
-        let icon: StyleKitIcon = (self.parent?.wr_splitViewController?.layoutSize == .compact) ? arrowIcon : .hamburger
+        let icon = backButtonIcon(hasUnreadInOtherConversations: false)
         let action = #selector(ConversationViewController.onBackButtonPressed(_:))
+
         let button = UIBarButtonItem(icon: icon, target: self, action: action)
         button.accessibilityIdentifier = "ConversationBackButton"
         button.accessibilityLabel = L10n.Accessibility.Conversation.BackButton.description
 
-        if hasUnreadInOtherConversations {
-            button.tintColor = UIColor.accent()
-            button.accessibilityValue = L10n.Localizable.ConversationList.Voiceover.UnreadMessages.hint
+        // `ZMConversation.hasUnreadMessageInOtherConversations` is expensive because it performs a fetch request,
+        // so we call it from a Task to reduce the load on the UI.
+        // If `true` we update the button icon, If `false` we already have set it up above
+        Task {
```

Comment:
Wouldn't it be cleaner to move this code into `leftNavigationItems(forConversation:)` or `updateLeftNavigationBarItems()`?

Reply:
@David-Henner I wonder if the task should be retained so if we hit this path rapidly we cancel the ongoing task for example

Reply:
that would need some refactoring since `backButton` is a computed property defined in an extension